vm/actor/src/util/set_multimap.rs

```diff
@@ -33,33 +33,35 @@ where
     }
 
     /// Puts the DealID in the hash set of the key.
-    pub fn put(&mut self, key: &Address, value: DealID) -> Result<(), String> {
+    pub fn put(&mut self, key: ChainEpoch, value: DealID) -> Result<(), String> {
         // Get construct amt from retrieved cid or create new
-        let mut set = self.get(key)?.unwrap_or_else(|| Set::new(self.0.store()));
+        let mut set = self
+            .get(u64_key(key))?
+            .unwrap_or_else(|| Set::new(self.0.store()));
 
         set.put(u64_key(value))?;
 
         // Save and calculate new root
         let new_root = set.root()?;
 
         // Set hamt node to set new root
-        Ok(self.0.set(key.to_bytes().into(), &new_root)?)
+        Ok(self.0.set(u64_key(key), &new_root)?)
     }
 
     /// Gets the set at the given index of the `SetMultimap`
     #[inline]
-    pub fn get(&self, key: &Address) -> Result<Option<Set<'a, BS>>, String> {
-        match self.0.get(&key.to_bytes())? {
+    pub fn get(&self, key: BytesKey) -> Result<Option<Set<'a, BS>>, String> {
```

Comment:
This should probably be &ChainEpoch or ChainEpoch, you shouldn't have to expose the underlying type for the public interface for this